Kernel/FileSystem/Ext2FileSystem.cpp

```diff
@@ -798,6 +798,19 @@ ErrorOr<NonnullRefPtr<Inode>> Ext2FS::get_inode(InodeIdentifier inode) const
     return new_inode;
 }
 
+ErrorOr<void> Ext2FSInode::compute_block_list_with_exclusive_locking()
+{
+    MutexLocker block_list_locker(m_block_list_lock);
+    if (m_block_list.is_empty())
+        m_block_list = TRY(compute_block_list());
+
+    if (m_block_list.is_empty()) {
+        dmesgln("Ext2FSInode[{}]::read_bytes(): Empty block list", identifier());
+        return EIO;
+    }
```

Comment:
Why is this here? it doesn't really have anything to do with computing the block list, so it should be part of read_bytes.